sdk/src/react/ui/components/_internals/action-button/components/NonOwnerActions.tsx

```
@@ -25,13 +33,56 @@ export function NonOwnerActions({
 	chainId,
 	orderbookKind,
 	lowestListing,
+	marketplaceType,
+	salesContractAddress,
+	salePrice,
+	quantityDecimals,
+	quantityRemaining,
 }: NonOwnerActionsProps) {
 	const { show: showBuyModal } = useBuyModal();
 	const { show: showMakeOfferModal } = useMakeOfferModal();
 
+	if (marketplaceType === 'shop') {
+		if (!salesContractAddress) {
+			throw new Error('salesContractAddress is required for SHOP card type');
+		}
+
+		return (
+			<ActionButtonBody
+				action={CollectibleCardAction.BUY}
+				tokenId={tokenId}
+				label="Buy now"
+				onClick={() =>
+					showBuyModal({
+						chainId,
+						collectionAddress,
+						salesContractAddress,
+						items: [
+							{
+								tokenId,
+								// This is overridden by quantity input state, see: sdk/src/react/ui/modals/BuyModal/hooks/useERC1155Checkout.ts
+								quantity: '1',
+							},
+						],
+						marketplaceType: 'shop',
+						salePrice: {
+							amount: salePrice?.amount ?? '',
+							currencyAddress: salePrice?.currencyAddress ?? '0x',
```

Same here this should be part of the discriminated union and not silently fallback to incorrect values